gateway/crates/config/src/entity_caching.rs

```diff
@@ -1,10 +1,62 @@
-use std::time::Duration;
+use std::{path::PathBuf, time::Duration};
 
 #[derive(Debug, Default, serde::Deserialize, Clone, PartialEq)]
 pub struct EntityCachingConfig {
     pub enabled: Option<bool>,
 
+    #[serde(default)]
+    pub storage: EntityCachingStorage,
+
+    #[serde(default)]
+    pub redis: Option<EntityCachingRedisConfig>,
+
     /// The ttl to store cache entries with.  Defaults to 60s
     #[serde(deserialize_with = "duration_str::deserialize_option_duration", default)]
     pub ttl: Option<Duration>,
 }
+
+#[derive(Debug, Clone, Default, PartialEq, serde::Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum EntityCachingStorage {
+    #[default]
+    Memory,
+    Redis,
+}
+
+#[derive(Debug, Clone, PartialEq, serde::Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct EntityCachingRedisConfig {
```

Comment:
:zap: **Suggestion** 
Consider adding a method to `EntityCachingRedisConfig` to validate the configuration, ensuring that the `url` is correctly formatted and the `key_prefix` is not empty, which can prevent runtime errors.